src/sentry/notifications/notificationcontroller.py

```diff
@@ -58,14 +61,24 @@ def __init__(
         organization_id: int | None = None,
         type: NotificationSettingEnum | None = None,
         provider: ExternalProviderEnum | None = None,
+        organization: Organization | None = None,
     ) -> None:
-        self.recipients = recipients
+        if features.has("organizations:team-workflow-notifications", organization):
+            self.recipients = []
+            for recipient in recipients:
+                if recipient_is_team(recipient):
+                    if provider and team_is_valid_recipient(recipient, provider):  # type: ignore  # recipient_is_team assures recipient type is okay
```

Comment:
@isabellaenriquez provider shouldn't be an input here as `team_is_valid_recipient` shouldn't depend on the provider.

Reply:
isn't whether a team is valid as a recipient contingent on whether they use slack as a provider and have a slack channel setup? otherwise what would we be checking for in the function?